analyzers/src/SonarAnalyzer.Common/Helpers/TypeHelper.cs

```diff
@@ -42,6 +42,9 @@ internal static class TypeHelper
         public static bool IsNullableValueType(this ITypeSymbol self) =>
             self.IsStruct() && self.OriginalDefinition.Is(KnownType.System_Nullable_T);
 
+        public static bool IsNonNullableValueType(this ITypeSymbol self) =>
```

Comment:
"non-nullable value type" is the name used in the C# spec:

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types#conversion-from-a-nullable-value-type-to-an-underlying-type